Orange/data/domain.py

```diff
@@ -58,69 +69,101 @@ def __init__(self, source, destination):
 
 
 class Domain:
-    def __init__(self, attributes, class_vars=None, metas=None, source=None):
+    def __init__(self, attributes=(), class_vars=(), metas=(), source=None):
```

Comment:
Should we move the indices/source logic to a separate class method, which converts indices to variables and constructs a new domain (by calling its init)?